streams/src/main/java/org/apache/kafka/streams/KafkaStreams.java

```diff
@@ -215,7 +245,9 @@ public synchronized void start() {
     public synchronized void close() {
         log.debug("Stopping Kafka Stream process");
 
-        if (state == RUNNING) {
+        if (state != KafkaStreamsState.PENDING_SHUTDOWN &&
```

Comment:
Change to check for is ("Created" or "Rebalancing" or "Running") -- a positive check is simpler to reason about than a reverted negative check.